test/sql/catalog/function/test_select_macro.test

```diff
@@ -0,0 +1,82 @@
+# name: test/sql/catalog/function/test_select_macro.test
+# description: Test SELECTMacro
+# group: [function]
+
+statement ok
+PRAGMA enable_verification
+
+
+statement ok
+CREATE TABLE test_tbl (id INT, name string);
+
+statement ok
+CREATE TABLE test2_tbl (id INT, name string);
+
+statement ok
+CREATE TABLE greek_tbl (id INT, name string);
+
+statement ok
+INSERT INTO test_tbl VALUES (1,'tom'), (2,'dick'),(3,'harry'), (4,'mary'), (5,'mungo'), (6,'midge');  
+
+statement ok
+INSERT INTO test_tbl VALUES (20,'andrew'), (21,'boris'),(22,'Caleb'), (23,'david'), (24,'evan');
+
+statement ok
+INSERT INTO  greek_tbl VALUES (1, 'alpha'), (2, 'beta'), (3, 'gamma'), (4, 'delta'), (5, 'epsilon'),(6, 'zeta'), (7, 'eta') , (8, 'theta'), (9, 'iota') , (10, 'kappa'); 
+
+statement ok
+CREATE MACRO TABLE xt(a,_name) as SELECT * FROM test_tbl WHERE(id>=a or name=_name);
+
+statement ok
+CREATE MACRO TABLE xt2(a,_name) as SELECT * FROM test_tbl WHERE(id>=a or name like _name);
+
+statement ok
+CREATE MACRO TABLE sgreek(a,b,c) as SELECT a,b FROM greek_tbl WHERE(id >= c);
```

Comment:
I would like to see many more tests, e.g.:

* Macro parameters in SELECT, GROUP BY, HAVING, QUALIFY, LIMIT, ORDER BY
* Macro parameters in scalar subqueries
* Macro parameters in regular subqueries
* Macro parameters that have the same name as columns in a table
* Name conflicts of table macros (i.e. we can't create a table macro called `generate_series` because there is already a table function of this name, but we can create a table macro called `sqrt` because there is no table function of this name).

Reply:
Yep more tests 
Out of interest where are the testing DB's found in the project